src/olympia/reviewers/tests/test_utils.py

```diff
@@ -1153,6 +1164,50 @@ def test_public_with_disabled_version_addon_confirm_auto_approval(self):
         # Check points awarded.
         self._check_score(amo.REVIEWED_EXTENSION_MEDIUM_RISK)
 
+    def test_addon_with_versions_pending_rejection_confirm_auto_approval(self):
+        self.grant_permission(self.request.user, 'Addons:PostReview')
+        self.grant_permission(self.request.user, 'Reviews:Admin')
+        self.setup_data(amo.STATUS_APPROVED, file_status=amo.STATUS_APPROVED)
+        self.version = version_factory(
+            addon=self.addon, version='3.0',
+            file_kw={'status': amo.STATUS_APPROVED})
+        self.file = self.version.files.all()[0]
+        summary = AutoApprovalSummary.objects.create(
+            version=self.version, verdict=amo.AUTO_APPROVED, weight=153)
+
+        for version in self.addon.versions.all():
+            VersionReviewerFlags.objects.create(
+                version=version,
+                pending_rejection=datetime.now() + timedelta(days=7))
+
+        self.helper = self.get_helper()  # To make it pick up the new version.
+        self.helper.set_data(self.get_data())
+
+        # We're an admin, so we can confirm auto approval even if the current
+        # version is pending rejection.
+        assert 'confirm_auto_approved' in self.helper.actions
+        self.helper.handler.confirm_auto_approved()
+
+        summary.reload()
+        assert summary.confirmed is True
+        approvals_counter = AddonApprovalsCounter.objects.get(addon=self.addon)
+        self.assertCloseToNow(approvals_counter.last_human_review)
+        assert self.check_log_count(amo.LOG.APPROVE_CONTENT.id) == 0
+        assert self.check_log_count(amo.LOG.CONFIRM_AUTO_APPROVED.id) == 1
+        activity = (ActivityLog.objects.for_addons(self.addon)
+                               .filter(action=amo.LOG.CONFIRM_AUTO_APPROVED.id)
+                               .get())
+        assert activity.arguments == [self.addon, self.version]
+        assert activity.details['comments'] == ''
+
+        # None of the versions should be pending rejection anymore.
+        assert not VersionReviewerFlags.objects.filter(
```

Comment:
As mentioned below (sorry, I reviewed the files in reverse order), if the data for this test only has one version pending rejection, would it be better to set up more than one version pending rejection to verify that they are all cleared? I suppose that doesn't need to be done in both tests, but is seems like there should be at least one test that verifies that all versions get cleared.

Reply:
That test has 2 versions pending rejection - one is created during the class `setUp()` with the addon, the other with the `version_factory()` call, and we loop over all available versions to set a pending rejection on both line 1178.